addons/api/addon/serializers/host-catalog.js

```diff
@@ -37,31 +58,28 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
 
   serializeAttribute(snapshot, json, key, attribute) {
     const value = super.serializeAttribute(...arguments);
-    const { isPlugin, compositeType } = snapshot.record;
+    const { compositeType, credentialType } = snapshot.record;
     const { options } = attribute;
 
-    // Delete any fields that don't belong to the compositeType
-    if (
-      isPlugin &&
-      json &&
-      options?.for &&
-      !options.for.includes(compositeType) &&
-      !fieldsByType[compositeType].includes(key)
-    ) {
-      delete json[key];
-    }
+    const fields =
+      compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE
+        ? azureFields
+        : AWSfieldsWithCredentialType[credentialType];
 
     // Delete any nested attribute fields that don't belong to the record type
-    if (isPlugin && options.isNestedAttribute && json.attributes) {
-      if (!fieldsByType[compositeType].includes(key)) {
+    if (options.isNestedAttribute && json.attributes) {
```

Comment:
How come we removed the `isPlugin` check?

Reply:
good catch, this should not be removed as the fields we check belong to plugin type